### PR TITLE
feat(clipboard): added wayclip support

### PR DESCRIPTION
**Pull request by XDream8**
wayclip -> https://git.sr.ht/~noocsharp/wayclip

### CODE REVIEW

**Comment by gpanders** [comment]
To satisfy the commit linter, use `feat(clipboard):` as the prefix.

Please also update the docs at `:h clipboard-tool`.

**Comment by XDream8** [comment]
> To satisfy the commit linter, use `feat(clipboard):` as the prefix.
> 
> Please also update the docs at `:h clipboard-tool`.

changed pr title and updated the docs

**Comment by gpanders** [comment]
The PR title doesn't matter. You need to update the commit message.

File: runtime/autoload/provider/clipboard.vim [hunk 1]
```diff
@@ -97,6 +97,12 @@ function! provider#clipboard#Executable() abort
     let s:copy['*'] = ['wl-copy', '--foreground', '--primary', '--type', 'text/plain']
     let s:paste['*'] = ['wl-paste', '--no-newline', '--primary']
     return 'wl-copy'
+  elseif !empty($WAYLAND_DISPLAY) && executable('waycopy') && executable('waypaste')
+    let s:copy['+'] = ['waycopy']
+    let s:paste['+'] = ['waypaste']
```
Comment (gpanders) [inline runtime/autoload/provider/clipboard.vim:102]: Does this need `-t text/plain` (like `wl-copy` and `wl-paste`)?

Comment (XDream8) [reply runtime/autoload/provider/clipboard.vim:102]: it does not need but can be added

Comment (gpanders) [reply runtime/autoload/provider/clipboard.vim:102]: Please add it for consistency.

Comment (XDream8) [reply runtime/autoload/provider/clipboard.vim]: added it!

**Pull request merged**